index.js

```diff
@@ -22,6 +22,10 @@ var PredictiveServiceClient = function(end_point, api_key, should_verify_certifi
   this.end_point = protocol_uri["uri"];
   this.protocol = protocol_uri["protocol"];
   this.timeout = 10000; // default to 10 seconds timeout
+  this.port = 80; // default port 80
```

Comment:
there's duplicated logic here.

remove line 21, 22, 23, 25, 26, 27, 28

replace them with:
```
this.setEndpoint(end_point);
```

If you want to initialize those instance variables at the top of the constructor, that's fine.